web-registry/src/features/ecocredit/CreateBatchBySteps/CreateBatchMultiStepForm/Result.tsx

```diff
@@ -0,0 +1,186 @@
+import React from 'react';
+import { DeliverTxResponse } from '@cosmjs/stargate';
+import { useNavigate } from 'react-router-dom';
+import { Box } from '@mui/material';
+
+import OnBoardingCard from 'web-components/lib/components/cards/OnBoardingCard';
+import {
+  // Body,
+  Label,
+  Subtitle,
+  Title,
+} from 'web-components/lib/components/typography';
+import OutlinedButton from 'web-components/lib/components/buttons/OutlinedButton';
+import { CardItem } from 'web-components/lib/components/modal/TxModal';
+import { BrokenLinkIcon } from 'web-components/lib/components/icons/BrokenLinkIcon';
+import { truncate } from 'web-components/lib/utils/truncate';
+
+import { Link } from '../../../../components/atoms';
+import { getAccountUrl, getHashUrl } from '../../../../lib/block-explorer';
+
+type ResultProps = {
+  response?: DeliverTxResponse;
+  error?: string;
+};
+
+export default function Result({
+  response,
+  error,
+}: ResultProps): React.ReactElement {
+  if (error) return <ErrorResult error={error} />;
+  return response ? <SuccessResult response={response} /> : <div />;
+}
+
+type SuccessProps = {
+  response: DeliverTxResponse;
+};
+
+const SuccessResult = ({ response }: SuccessProps): React.ReactElement => {
+  const navigate = useNavigate();
+
+  // Parsing the response...
+  const responseLog = response?.rawLog && JSON.parse(response?.rawLog);
+  const responseLogEvents = responseLog && responseLog[0].events;
+
+  const eventCreateBatch =
+    responseLogEvents &&
+    responseLogEvents.find((event: any) =>
+      event.type.includes('.EventCreateBatch'),
+    );
+
+  const receiveBatchDenom =
+    eventCreateBatch &&
+    eventCreateBatch.attributes?.find((obj: any) => obj.key === 'batch_denom');
+
+  const eventReceive =
+    responseLogEvents &&
+    responseLogEvents.find((event: any) =>
+      event.type.includes('.EventReceive'),
+    );
+
+  const recipientsLog =
+    eventReceive &&
+    eventReceive.attributes?.filter((obj: any) => obj.key === 'recipient');
+  const recipients = recipientsLog.map(({ value }: { value: string }) => {
```

Comment:
wouldn't that be possible to use the recipients addr values from the form state in here instead of parsing the response?

Reply:
@blushi  While that would be easier, I think the intent here is to explicitly display content from the response, for verification purposes. Hard to imagine how they could differ, but I think it makes sense to do it this way.

Reply:
the only thing is that it's just more dependent on the ledger API so could be more sensitive to future breaking changes (although I think at this point, it should stay quite stable) but fine by me to keep it this way.
I was just pointing this because for the other txs (send, retire, etc.), we use the form values for now to display on the confirmation page. I think it'd be good to unify this at some point, just for consistency.